src/de/map.rs

```diff
@@ -135,7 +140,24 @@ enum ValueSource {
     Nested,
 }
 
-/// A deserializer for `Attributes`
+/// A deserializer that extracts map-like structures from an XML. This deserializer
+/// represents a one XML tag:
+///
+/// ```xml
+/// <tag>...</tag>
+/// ```
+///
+/// Name of this tag is stored in a [`Self::start`] property.
+///
+/// # Lifetimes
+///
+/// - `'de` lifetime represents a buffer, from which deserialized values can
+///   borrow their data. Depending on the underlying reader, there can be an
+///   internal buffer of deserializer (i.e. deserializer itself) or an input
+///   (in that case it is possible to approach zero-cost deserialization).
```

Comment:
Zero-cost, or zero-copy?  The latter is what is specifically being achieved here while the former is kind of a fuzzy statement.  I take it to mean that "deserialization is fast enough that the CPU can keep up with IO" in the same vein as e.g. `simdjson`

Reply:
I mean that when optimized any serde deserializer code should gone as if you manually locate positions in the string and take a slice. But `zero-copy` is a better anyway.